listenbrainz/webserver/static/js/src/BrainzPlayer.tsx

```diff
@@ -247,13 +244,7 @@ export default class BrainzPlayer extends React.Component<
 
     let nextListenIndex;
     if (currentListenIndex === -1) {
-      nextListenIndex = direction === "up" ? listens.length - 1 : 0;
-    } else if (direction === "up") {
-      nextListenIndex =
-        invert === true ? currentListenIndex + 1 : currentListenIndex - 1 || 0;
-    } else if (direction === "down") {
-      nextListenIndex =
-        invert === true ? currentListenIndex - 1 || 0 : currentListenIndex + 1;
+      nextListenIndex = 0;
     } else {
       this.handleWarning("Please select a song to play", "Unrecognised state");
       return;
```

Comment:
I tried that, but it doesnt seem to like it, I had a bit of an attempt at fixing it by doing this:
```
    if (currentListenIndex === -1) {
      nextListenIndex = 0;
    } if (invert === true) {
      nextListenIndex = invert === true ? currentListenIndex - 1 || 0 : currentListenIndex + 1;
    } else {
      this.handleWarning("Please select a song to play", "Unrecognised state");
      return;
    }
```

Thinking of it though, it wouldn't make sense to only increment / decrement the index on the `invert === true condition`. A suitable if condition could make it work though lol.

Reply:
What do you mean by "but it doesnt seem to like it" ?
If you'r referring to the failing test, it's unrelated; the tests are failing on a formatting issue: https://github.com/metabrainz/listenbrainz-server/runs/4193409065?check_suite_focus=true#step:9:23

We run a linter to check that the code is format in a certain way.
You can run it yourself locally before committing changes by running `./test.sh fe -f` to run the linter in a Docker container.
Alternatively if you had better luck installing Node dependencies locally, you can run `npm run format` directly. It's what the docker method also does, but in its own environment.

All that being said, I rewrote the code from my comment above to be a bit more explicit.
The block should be this:
```
    let nextListenIndex;
    if (currentListenIndex === -1) {
      // No current listen index found, default to first item
      nextListenIndex = 0;
    } else if (invert) {
      // `|| 0` constrains to positive numbers
      nextListenIndex = currentListenIndex - 1 || 0;
    } else {
      nextListenIndex = currentListenIndex + 1;
    }
```

After that, run the linter to automatically format, and you should be good to commit :)


P.S: In case you are wondering, you can view failing test details by clicking the red "x" next to a commit and then the "Details" button , in this case for front-end tests:
![image](https://user-images.githubusercontent.com/6179856/141776498-9abbdf31-fe2e-4b7c-99e2-6ea9c5d0840a.png)

Reply:
Oh, my mistake, I thought that there was another failed test, but it was just the test from the previous commits. I see that linting issue now! I've pushed the new fix.. will see how that goes